src/typeahead/index.js

```diff
@@ -88,13 +88,16 @@ var Typeahead = React.createClass({
       selection: this.props.value,
 
       // Index of the selection
-      selectionIndex: null
+      selectionIndex: null,
+
+      isFocused: false,
     };
   },
 
   _shouldSkipSearch: function(input) {
     var emptyValue = !input || input.trim().length == 0;
-    return !this.props.showOptionsWhenEmpty && emptyValue;
+    var isFocused = this.state && this.state.isFocused;
```

Comment:
It's kind of funky, but the reason is that `_shouldSkipSearch` is called from `getOptionsForValue`, which is called from `getInitialState`. Within `getInitialState`, `this.state` is not yet defined, so this has to be special cased.